src/gt4py/next/constructors.py

```diff
@@ -189,7 +189,10 @@ def full(
         allocator=allocator,
         device=device,
     )
-    field[...] = field.dtype.scalar_type(fill_value)
+    if hasattr(fill_value, "__array__"):
+        field[...] = field.array_ns.asarray(fill_value)
+    else:
+        field[...] = field.dtype.scalar_type(fill_value)
```

Comment:
I don't think this is the right solution. `full`, by definition, should only accept scalar fill values. If the provided `fill_value` is an ndarray and not a scalar, then the `as_field()` function should be used.